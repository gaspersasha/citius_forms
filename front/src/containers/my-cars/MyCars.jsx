import React, { useState, useEffect } from 'react';
import { Spinner } from '~components';
import { AccountVehicleList } from '~containers';
import { endpointFetch } from '~utils';
import { ArrowRight, NoSavedCars } from '~assets/svg';
import s from './styles/my-cars-container.module.sass';

export default function MyCars() {
  const [vehicles, setVehicles] = useState([]);
  const [allVehicles, setAllVehicles] = useState(false);
  const [isLoading, setLoading] = useState(true);

  useEffect(() => {
    endpointFetch('myCars', {
      limit: 7,
      offset: 1,
    })
      .then((data) => {
        setVehicles(data?.savedVehicles);
        setAllVehicles(data?.allVehicles);
      })
      .catch(console.error)
      .finally(setLoading);
  }, []);

  if (isLoading) {
    return <Spinner className={s.spinner} />;
  }

  return (
    <div className={s.container}>
      <h1 className="page-title">My Cars</h1>
      {vehicles?.length ? (
        <>
          <AccountVehicleList items={vehicles} allVehicles={allVehicles} />
          <div className={s.moreCars}>
            <p className={s.moreCarsTitle}>Still not found your perfect car?</p>
            <a href="/cars" className={s.link}>
              Continue my car search
            </a>
          </div>

          <div className={s.footnote}>
            <div className={s.footnoteContainer}>
              <b>Representative example when buying on PCP</b>
              <p>
                Borrowing £9,500 over 48 months, zero deposit, on type PCP, an
                annual mileage of 8,000pa, with a Representative APR of 9.9%,
                the amount payable would be £191.54 a month, an optional final
                payment of £2923.26, with a total cost of credit of £2,425.64
                and a total amount payable of £11,925.64 (** based on a 2017
                Ford Fiesta 1.0 TITANIUM X Hatchback)
              </p>
            </div>
          </div>
        </>
      ) : (
        <div className={s.noCars}>
          <div className={s.content}>
            <p className={s.noCarsTitle}>No saved cars</p>
            <p className={s.noCarsSubtitle}>Your car search starts here</p>
            <ul className={s.list}>
              <li className={s.listItem}>Search for a car</li>
              <li className={s.listItem}>Save your favorites</li>
              <li className={s.listItem}>Apply for finance or buy outright</li>
            </ul>
            <a href="/cars" className={s.CTA}>
              Find your next car
              <ArrowRight className={s.icon} />
            </a>
          </div>
          <NoSavedCars className={s.media} />
        </div>
      )}
    </div>
  );
}
