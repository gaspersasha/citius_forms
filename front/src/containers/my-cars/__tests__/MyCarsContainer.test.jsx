import React from 'react';
import { endpointFetch } from '~utils';
import MyCars from '../MyCars';

jest.mock('~utils', () => ({
  ...jest.requireActual('../../../utils'),
  endpointFetch: jest.fn(),
}));

endpointFetch.mockResolvedValue(global.mockData.myCars(6));

let wrapper;

beforeEach(() => {
  wrapper = shallow(<MyCars />);
});

describe('<MyCars />', () => {
  it('should match snapshot', async () => {
    await global.updateWrapper(wrapper);
    expect(wrapper).toMatchSnapshot();
  });

  it('placeholder view should display if there is no cars fetched from server', async () => {
    endpointFetch.mockResolvedValue(null);
    wrapper = mount(<MyCars />);
    await global.updateWrapper(wrapper);
    const placeholder = wrapper.find('NoSavedCars');

    expect(placeholder.exists()).toBe(true);
  });
});
