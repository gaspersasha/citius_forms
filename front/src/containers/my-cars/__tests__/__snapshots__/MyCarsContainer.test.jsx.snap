// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyCars /> should match snapshot 1`] = `
<Spinner
  className="spinner"
/>
`;
